traitsui/qt4/button_editor.py

```diff
@@ -75,6 +75,19 @@ def init(self, parent):
     def dispose(self):
         """ Disposes of the contents of an editor.
         """
+
+        if self.factory.values_trait:
```

Comment:
There are also connections made between `action.triggered` and a lambda function in `_update_menu`. They will be disconnected when `action` objects are destroyed. I wonder if it's worth adding explicit disconnects for the case when the objects aren't destroyed for some reason.

Reply:
Considering that the feature does not seem to work, I am leaning towards the answer "not worth it". See https://github.com/enthought/traitsui/issues/879